### DISCUSSION TITLE
Création d'un acteur via script

### DISCUSSION THREAD

**Discussion by KenneyClement** [witsa/synapps#785] [Q&A] [answered]
J'aimerai créer un script qui créer un acteur  lors de l'initialisation d'un composite mais le liens de l'image ne passe pas 

context.getActor('stackEXT').addActors(
    [{
       "children": [
       {
          "type": "display/image",
          "key": "plug1",
          "bindings": {
            "properties.content": "picture@PLUG501"
          }
      }]
    }]);
    
    SYNAPPS-STUDIO-COMPOSITE|{"config":{"key":"composite1","name":"Composite 1"},"leadActor":{"type":"layout/stack","key":"stack1","children":[{"type":"redy/data-source/wos-variable","key":"wos-variable1","properties":{"path":":easy.RESS.OwnerPLUG","fieldName":"children","autoRefreshDelay":5},"additionalDefs":{"path":{"type":"wos-path"}},"bindings":{"properties.path":"stage@properties.Path"},"events":{"onInit":["","context.getActor('stackEXT').addActors(","    [{","       \"children\": [","       {","          \"type\": \"display/image\",","          \"key\": \"plug1\",","          \"bindings\": {","            \"properties.content\": \"picture@PLUG501\"","          }","      }]","    }]);"]}},{"type":"layout/stack","key":"stackEXT","properties":{"orientation":"horizontal"}}],"properties":{"height":"200px"}}}

**Comment by manufitoussiwit** [comment, marked as answer]
La définition de la liaison doit être complète, pas en version résumé. Cette dernière n'est pas comprise par le Runtime. 
```js
context.getActor('stackEXT').addActors([
  {        
    type: "display/image",
    key: "plug1",
    bindings: {
      "properties.content": {
        sourceType: "picture",
        path: PLUG501"
      }
    }
  }
]);
```
J'ai corrigé aussi la syntaxe attendu par `addActors`.

Pour un seul acteur, on peut utiliser aussi `addActor` : 

```js
context.getActor('stackEXT').addActor({        
  type: "display/image",
  key: "plug1",
  bindings: {
    "properties.content": {
      sourceType: "picture",
      path: PLUG501"
    }
  }
});
```
